src/cascadia/TerminalApp/Profile.cpp

```diff
@@ -494,15 +496,16 @@ bool Profile::HasIcon() const noexcept
 // - tabTitle: the tab title
 void Profile::SetTabTitle(std::wstring tabTitle) noexcept
 {
-    _tabTitle = tabTitle;
+    _tabTitle = std::move(tabTitle);
 }
 
 // Method Description:
 // - Sets this profile's icon path.
 // Arguments:
 // - path: the path
-void Profile::SetIconPath(std::wstring_view path) noexcept
+void Profile::SetIconPath(std::wstring_view path)
 {
+    static_assert(!noexcept(_icon.emplace(path)));

```

Comment:
Related to the above, what's this doing and why does this one have a `!` in it?

Reply:
It's the flip side of the previous comment. Maybe it'd be clearer if I'd had the comment for each static_assert:
```
static_assert(noexcept(_schemeName = std::move(schemeName)), "Profile::SetColorScheme() is no longer noexcept!")
static_assert(!noexcept(_icon.emplace(path)), "Mark Profile::SetIconPath() as noexcept");

```

Reply:
Yea a comment might be helpful, since I definitely read that code and assumed that it was asserting the opposite :P